### PR TITLE
chore: add unit test coverage and enforcement

### PR DESCRIPTION
**Pull request by ssvegaraju**
Issue #, if available:
FHIR-545

Description of changes:
Added workflow for testing code cov and set limit to 0 initially to see what the coverage is at before choosing a limit. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by nguyen102** [comment]
You can check the code coverage by running `yarn test-coverage` locally.

I had to run this to check for the code cov of FHIR repos a couple of weeks ago. You can see my results [here](https://quip-amazon.com/8cKYAMzb0tqM/Chamonix-How-We-Work#temp:C:MTZ725c071e2d4db47a9fcd7d987)

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+coverage:
+  round: down
+  precision: 2
+  range: '0...100'
+  status:
+    project:
+      default:
+        # basic
+        target: 0%
```
Comment (nguyen102) [inline codecov.yml]: In the How We Work, I recommended 90% for this. All of our FHIR Works repo have more than 90% coverage.

Comment (ssvegaraju) [reply codecov.yml]: That's super helpful, thanks! Was originally just trying to see what coverage we were at but it didn't occur to me to just use yarn.

File: codecov.yml [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
   round: down
   precision: 2
-  range: '0...100'
+  range: '90...100'
   status:
     project:
       default:
         # basic
-        target: 0%
+        target: 00%
```
Comment (nguyen102) [inline codecov.yml]: Should this be 90? That's the target we're aiming for.

File: .github/workflows/code-coverage.yml [hunk 3]
```diff
@@ -5,7 +5,7 @@
 
 name: Code Coverage Report
 
-on: [push, pull_request]
+on: [push]
```
Comment (nguyen102) [inline .github/workflows/code-coverage.yml]: I think we want the code coverage to run on pull requests as well, so the developer knows if their code coverage meets the guideline before they merge to develop

File: codecov.yml (hunk 2 above)
Comment (ssvegaraju) [reply codecov.yml]: Yes, that's a typo 😅

File: .github/workflows/code-coverage.yml (hunk 3 above)
Comment (ssvegaraju) [reply .github/workflows/code-coverage.yml]: That makes sense, I'll put that back in.

**Review by nguyen102 — APPROVED** [review]

**Pull request merged**